pkg/inject/report.go

```diff
@@ -51,7 +70,7 @@ func (r *Report) Injectable() bool {
 
 // update updates the report for the provided resource conf.
 func (r *Report) update(conf *ResourceConfig) {
-	r.InjectDisabled = conf.pod.Meta.GetAnnotations()[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled
+	r.InjectDisabled = conf.pod.meta.GetAnnotations()[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled || (conf.nsAnnotations[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled && conf.pod.meta.GetAnnotations()[k8s.ProxyInjectAnnotation] != k8s.ProxyInjectEnabled)
```

Comment:
This is a really long line now. And I think there's an issue with the logic here, since I'm seeing the auto-injector from this branch inject pods in namespaces without the `linkerd.io/inject: enabled` annotation. It is correctly skipping pods with the `linkerd.io/inject: disabled` annotation. The issue only arises when the namespace has no `linkerd.io/inject` annotation.

Reply:
Yes, you are right about the regression, when the `linkerd.io/inject` annotation isn't present. Sorry about the long one liner. I will break it up into a function or something.

Reply:
The issue is that this conditional didn't account for the different implication in the absence of the `linkerd.io/inject` annotation in the webhook and the CLI. In the webhook, it implies an opt-out, while with `linkerd inject`, it implies an opt-in. I have fixed it in the `HEAD` of this branch.